deltaspike/modules/partial-bean/impl/src/test/java/org/apache/deltaspike/test/core/api/partialbean/uc013/MethodLevelCacheInterceptorTest.java

```diff
@@ -75,17 +66,14 @@ public static WebArchive war()
     @Test
     public void testMethodLevelInterceptor() throws Exception
     {
-        // this test is known to not work under weld-2.0.0.Final and weld-2.0.0.SP1
-        Assume.assumeTrue(!CdiContainerUnderTest.is(CONTAINER_WELD_2_0_0));
-
         MyRepository myRepository = BeanProvider.getContextualReference(MyRepository.class);
 
         List<String> users = myRepository.getAllUsers();
         
         Assert.assertNotNull(users);
         Assert.assertEquals(3, users.size());
 
-        Assert.assertSame(users, myRepository.getAllUsers());
+        Assert.assertEquals(users, myRepository.getAllUsers());

```

Comment:
IMO assertSame is by design here as the result is cached - it needs to be the same object instance

Reply:
on Wildfly-managed it is not, as it seems - or else no fail. 
IMHO it is up to the container to serialize/deserialize request/response objects down the interceptor chain - whatever it may be.
Iff this poses a problem, I could try to identify where this - evident? - ser/deser happens to better understand what to do.
Another possibilty would be to check for "jboss.server.name" SysProp here to capture the case.
Any decision welcome!

Reply:
There is no serialization here
The interceptor caches the method return value
If it doesnt, its still broken for weld
Therefore the exclude propably

Reply:
Is it also broken with deltaspike.bean-manager.delegate_lookup?